zulipterminal/ui_tools/boxes.py

```diff
@@ -1112,22 +1112,34 @@ def reactions_view(self, reactions: List[Dict[str, Any]]) -> Any:
         if not reactions:
             return ""
         try:
-            reaction_stats = defaultdict(set)
+            my_user_id = self.model.user_id
+            reaction_stats = defaultdict(list)
             for reaction in reactions:
                 user_id = int(reaction["user"].get("id", -1))
                 if user_id == -1:
                     user_id = int(reaction["user"]["user_id"])
-                reaction_stats[reaction["emoji_name"]].add(user_id)
+                user_name = reaction["user"]["full_name"]
+                if user_id == my_user_id:
+                    user_name = "You"
+                reaction_stats[reaction["emoji_name"]].append((user_id, user_name))
 
             sorted_stats = sorted(
```

Comment:
Do we need this, now that the tests also pass and given that ordering is also preserved? This would show emojis in the order in which they were reacted. This can be a separate commit.